### ISSUE TITLE
can't save vo-table with text columns

### ISSUE THREAD

**Issue by yannick1974** [atpy/atpy#31]
Hi,

I'm using atpy to add IAU names to some vo-tables. To do so, I create and empty column with (catalogue is the atpy.Table object):

``` python
catalogue.add_empty_column(
    'iau_name', 
    description="IAU name", 
    dtype='S36', 
    position=0)
```

and populate it calculating the IAU names.

When I save the table to FITS, the new column is correctly save, but if I save it to vo-table, I have this error from python-vo:

``` python
?:?:?: W47: Missing arraysize indicates length 1
```

The arraysize of the column is the to 1 and the column is empty.

The problem occurs also when reading a vo-table (or a fits) with a text column, saving it discard all the textual columns.

Yannick

**Comment by astrofrog** [comment]
Thanks for reporting this. I can't reproduce the issue with a simple table, so I was wondering whether you can send the full script you are using to thomas.robitaille@gmail.com to see if I can reproduce the issue?

**Comment by b-1887** [comment]
I've come across the same problem, while using the latest version of atpy, vo 0.7.x, and Ubuntu 11.10. 
I can recreate the problem by creating a table, adding a set of string columns, such as:

t = atpy.Table() 
t.add_column('test1', numpy.array(['val1','val2','val3'])

When attempting to debug, I noticed in the votable.py file at line 197 approximately, where the assignment table.array[name] = self.data[name].astype(np.object_) is made, printing out the the second part of the assignment correctly prints out all the values, while printing the table array only displays the first character of every entry.

I guess this may have to do with different types (ndarray, recarray, etc ) but I'm not sure, as I haven't played around with numpy at all. The only fix I could figure out was to manually change the arraysize variable to some value e.g. 100, which would then however only display up to 100 characters of each entry.

Not sure if any of this helps!

An additional error I encountered, was when attempting to create a column of unicode characters:

raise Exception("cannot use numpy type " + str(column_type))
Exception: cannot use numpy type <type 'numpy.unicode_'>,  
which I guess is just a matter of adding another type to the type dict in votable.py

**Comment by astrofrog** [comment]
I'm very sorry for the delay in getting back to you about this bug. This is due to a change in behavior in the vo module. I've fixed ATpy in 99a553804fe5aed231317f8aca60c453f29a3da0 - could you try updating to the latest version to let me know if the issue goes away?

**Comment by yannick1974** [comment]
Hi Thomas.  At least, I took time to check this new version.  It now works as expected.
Thanks.
Yannick
